File: DSA/PriyankaSingh_590016827/day6question2.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>

// Function to check if array is strictly increasing
bool isStrictlyIncreasing(int* nums, int numsSize) 
{
    for (int i = 1; i < numsSize; i++) 
    {
        if (nums[i] <= nums[i-1]) 
        {
            return false;
        }
    }
    return true;
}

int main() 
{
    int n;
    
    printf("Enter the number of elements in the array: ");
    scanf("%d", &n);
    
    if (n <= 0) 
    {
        printf("Invalid array size!\n");
        return 1;
    }
    
    int* nums = (int*)malloc(n * sizeof(int));
    printf("Enter the elements of the array:\n");
    for (int i = 0; i < n; i++) 
    {
        scanf("%d", &nums[i]);
    }
    
    bool result = isStrictlyIncreasing(nums, n);
    
    if (result) 
    {
        printf("The array is strictly increasing.\n");
    } 
    else 
    {
        printf("The array is NOT strictly increasing.\n");
    }
    
    free(nums);
    return 0;
}

File: DSA/Pranvkumar_590011587/Day6Question2.py
def is_strictly_increasing(nums):
    for i in range(1, len(nums)):
        if nums[i] <= nums[i-1]:
            return False
    return True

# Example :
if __name__ == "__main__":
    print(is_strictly_increasing([1, 2, 3, 4]))   
    print(is_strictly_increasing([1, 2, 2, 3]))  
    print(is_strictly_increasing([5, 4, 3]))       
